bluesky/callbacks/fitting.py

```diff
@@ -281,8 +281,8 @@ def compute(self):
         if _cen_list:
             self.cen = np.mean(_cen_list)
 
-        if len(_cen_list) == 2:
-            self.fwhm = float(_cen_list[1] - _cen_list[0])
+        if len(_cen_list) >= 2:
```

Comment:
The commit message labels this a bug fix. Was `== 2` a bug? Itseems designed to silence the FWHM computation if the half-max crossings aren't clean and obvious.

Reply:
I think implementing this way not only covers the previous case when _cen_list == 2, but also extends the capability to deal with noisy data where crossings aren't clean and obvious, or more than 2. For noisy data, FWHM is difficult to get but this calculation can at least provide an initial guess.

Reply:
OK, that's reasonably compelling.

Reply:
Could we second a second pair of eyes on this before merging? Seem OK @mrakitin @jrmlhermitte  @tacaswell? Too brittle?

Reply:
I like this implementation when we consider only external points better. As I mentioned a few months ago, eventually we may come up with two different definitions of FWHM:
- local FWHM, which will essentially be a list of FWHM values for each found peak (need a reliable peaks finder, e.g. https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.signal.find_peaks_cwt.html)
- global FWHM, which will calculate the width between two external peaks. Let's devote a separate PR for that.